src/System.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs

```diff
@@ -32,6 +40,66 @@ public static bool AreConnStringsSetup()
             return !string.IsNullOrEmpty(NpConnStr) && !string.IsNullOrEmpty(TcpConnStr);
         }
 
+        public static bool IsServiceBrokerEnabled()
+        {
+            if (!serviceBrokerEnabled.HasValue)
+            {
+                serviceBrokerEnabled = false;
+                if (AreConnStringsSetup())
+                {
+                    try
+                    {
+                        var builder = new SqlConnectionStringBuilder(TcpConnStr);
+                        string database = builder.InitialCatalog;
+                        builder.ConnectTimeout = 2;
+                        using (var connection = new SqlConnection(builder.ToString()))
+                        using (var command = new SqlCommand("SELECT is_service_broker_enabled FROM sys.sys.databases WHERE name=@name", connection))
+                        {
+                            command.Parameters.AddWithValue("name", database);
+                            using (var reader = command.ExecuteReader())
+                            {
+                                if (reader.HasRows && reader.Read())
+                                {
+                                    serviceBrokerEnabled = (reader.GetInt32(0) == 1);
+                                }
+                            }
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        serviceBrokerEnabled = false;
+                    }
+                }
+            }
+            return serviceBrokerEnabled.Value;
+        }
+
+        public static bool IsDatabasePresent(string name)
+        {
+            databasesAvailable = databasesAvailable ?? new Dictionary<string, bool>();
+            bool present = false;
+            if (AreConnStringsSetup() && !string.IsNullOrEmpty(name) && !databasesAvailable.TryGetValue(name, out present))
+            {
+                try
+                {
+                    var builder = new SqlConnectionStringBuilder(TcpConnStr);
+                    builder.ConnectTimeout = 2;
+                    using (var connection = new SqlConnection(builder.ToString()))
+                    using (var command = new SqlCommand("SELECT COUNT(*) FROM sys.databases WHERE name=@name", connection))
+                    {
+                        connection.Open();
+                        command.Parameters.AddWithValue("name", name);
+                        present = Convert.ToInt32(command.ExecuteScalar()) == 1;
+                    }
+                }
+                catch (Exception)
+                {
```

Comment:
You should probably Console.WriteLine the error message here so that it is possible to find out why this setup retrieval failed.

Reply:
On second thoughts, why not let this exception surface to the test runner itself? Any exception here could be because of 
1. Product bug
2. Because of network problems trying to reach SQL Server. 
In either of those two cases, it might be futile to go ahead with the test.

Reply:
Done.